libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -700,21 +701,24 @@ class MapboxNavigation(
          * @param feedbackSource one of [FeedbackEvent.Source]
          * @param screenshot encoded screenshot (optional)
          * @param feedbackSubType array of [FeedbackEvent.Description] (optional)
+         * @param appMetadata [AppMetadata] information (optional)
          */
         @JvmStatic
         fun postUserFeedback(
             @FeedbackEvent.Type feedbackType: String,
             description: String,
             @FeedbackEvent.Source feedbackSource: String,
             screenshot: String?,
-            feedbackSubType: Array<String>? = emptyArray()
+            feedbackSubType: Array<String>? = emptyArray(),
+            appMetadata: AppMetadata? = null
```

Comment:
why are we sending app metadata every time we post user feedback?

IMO something like app metadata should be accessible anywhere in the app. We shouldn't have to pass it to every feature that needs it.

Reply:
Are we supporting the ability to postUserFeedback for different users? or different applications?

Reply:
The minimal app name information like user agent is already available in all telemetry events. This is only targeted at the feedback event to add additional context or match the event with other data that we have available to resolve the issue (like history traces from 1TAP). Any app developer that doesn't use the drop-in UI posts user feedback manually. We can take advantage of the field in our internal apps or work with other customers to enhance the data.

Reply:
Ok agreeing with the reason to have the data. Wondering why this is part of posting feedback, rather than app configuration. This interface supports a single app and user giving feedback for multiple apps and multiple users

Reply:
The metadata is dynamic, the session IDs can change during the lifecycle of the app and one feedback event can have a different session ID than the other. It could be set globally but now the schema is designed to accept the metadata for each feedback event.

Reply:
it sounds like the reason we need to send this object to postUserFeedback, is to send the session-id. We could get the app data from somewhere else (BuildConfig.APPLICATION_ID, BuildConfig.VERSION_NAME)

It doesn't sound like there is a reason to make those configurable from here. So it is confusing to have two required parameters, that are not configurable. But it was mentioned in a few places, will leave it up to you decide 👍

Reply:
We can tackle whatever is needed as part of https://github.com/mapbox/navigation-sdks/issues/284.